### PR TITLE
[baremetal] Adding the baremetal extension to the overall cli extension repo

### PR DESCRIPTION
**Pull request by pabowers**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [X] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
baremetal

**Review by AprilZhaoWY — APPROVED** [review]

File: src/index.json [hunk 1]
```diff
@@ -3480,6 +3480,52 @@
                 "sha256Digest": "bbf330942daac0035bd9bb9a0fdeaf297124ec8cd98e28e46adb2bf34070295b"
             }
         ],
+        "baremetal": [
```
Comment (AprilZhaoWY) [inline src/index.json:3483]: Please rename this to something like `baremetal-infrastructure` to be specific

Comment (pabowers) [reply src/index.json:3483]: Will do, but I believe that I will need to make a new release on our CLI extension to reflect this naming change

Comment (pabowers) [reply src/index.json]: I have updated in both places now, so it should be good

**Review by haroldrandom — APPROVED** [review]

File: src/index.json [hunk 2]
```diff
@@ -3480,6 +3480,52 @@
                 "sha256Digest": "bbf330942daac0035bd9bb9a0fdeaf297124ec8cd98e28e46adb2bf34070295b"
             }
         ],
+        "baremetal-infrastructure": [
```
Comment (haroldrandom) [inline src/index.json:3483]:
```suggestion
        "baremetal": [
```

I think the name should be the same as the name property value of `extensions`

**Review by haroldrandom — CHANGES_REQUESTED** [review]
Please fix the name of the index entry

**Comment by haroldrandom** [comment]
Hi, @pabowers @AprilZhaoWY! Nice to meet you. I believe it's the first time we meet in Azure CLI Extensions. I don't know whether you have contacted other ones on the team and they know the context or not.

Anyway, could you show us some context about your extension?

**Review by haroldrandom — APPROVED** [review]

**Pull request merged**